proposals/utils/pdf_diff_logic.py

```diff
@@ -0,0 +1,1134 @@
+            return mark_safe(self.object._meta.get_field(field).verbose_name)
+        else:
+            return mark_safe(self.object._meta.get_field(field).verbose_name % self.object.net_duration())
+
+class RowValueClass:
+    '''The RowValueClass manages the values of fields and correctly retrieves and/or formats
+    the right values per field. This class get initiated in the value method
+    of the RowClass. It returns mostly strings, but sometimes some html as well.'''
+
+    def __init__(self, object, field):
+
+        self.object = object
+        self.field = field
+
+    def get_field_value(self):
+        from ..models import Relation
+        from studies.models import Compensation
+        
+        value = getattr(self.object, self.field)
+
+        User = get_user_model()
+
+        if value in ('Y', 'N', '?'):
+            return self.yes_no_doubt(value)
+        elif isinstance(value, bool):
+            return _('ja') if value else _('nee')
+        elif isinstance(value, (str, int, date)):
+            return value
+        elif value is None:
+            return _('Onbekend')
+        elif isinstance(value, User):
+            return self.handle_user(value)
+        elif isinstance(value, Relation) or isinstance(value, Compensation):
+            return value.description
+        elif value.__class__.__name__ == 'ManyRelatedManager':
+            if value.all().model == User:
+                return self.get_applicants_names(value)
+            else:
+                return self.get_object_list(value)
+        elif value.__class__.__name__ == 'FieldFile':
+            return self.handle_field_file(value)
+        elif callable(value):
+            return value()
+        
+    def handle_user(self, user):
+        return user.get_full_name()
+    
+    def get_applicants_names(self, applicants):
+        applicant_names = [applicant.get_full_name() for applicant in applicants.all()]
+        return self.create_unordered_html_list(applicant_names)
+    
+    def get_object_list(self, object):
+        list_of_objects = [obj for obj in object.all()]
+        return self.create_unordered_html_list(list_of_objects)
+    
+    def create_unordered_html_list(self, list):
+        html_output = mark_safe('<ul class="p-0">')
+
+        for item in list:
+            html_output += format_html('<li>{}</li>',
+                item
+            )
+        
+        html_output += mark_safe('</ul>')
+
+        return html_output
+    
+    def handle_field_file(self, field_file):
+        if field_file:
+            output = format_html('<a href={}>{}</a>',
+                f'{settings.BASE_URL}{field_file.url}',
+                _('Download')
+            )
+        else:
+            output = _('Niet aangeleverd')
+
+        return output
+    
+    def yes_no_doubt(self, value):
+        from main.models import YES_NO_DOUBT
+        d = dict(YES_NO_DOUBT)
+        return d[value]
+    
+class StudyTitleClass:
```

Comment:
I think this should be called `SubTitle` or something; It's only used in `get_sub_title`, so it would be nice if the name matches up